### ISSUE TITLE
[Bug] Horns stuck on "generating draft" bit.

### ISSUE THREAD

**Issue by Mieridduryn** [dreadhalor/hermitcraft-horns#22]
What it says on the tin - when attempting to generate a horn (Firefox, Opera tried), after you hit "Generate", it gets stuck on "Generating..." for eternity.

**Comment by UplandJacob** [comment]
This was discussed a while back in #19. I didn't create an issue at the time because Dreadhalor knew about it and just hasn't gotten back to it. Good to keep track of it now though.

**Comment by ArtemiyStark** [comment]
> What it says on the tin - when attempting to generate a horn (Firefox, Opera tried), after you hit "Generate", it gets stuck on "Generating..." for eternity.

yeah, it is like that for 8 months already (。﹏。)

**Comment by dreadhalor** [comment]
Yep, unfortunately there doesn't seem to be a good way to solve it. I got close to a solution by wrapping the server with NordVPN & then getting around the YouTube API's ban on EC2 instance IP addresses, but the configs got really complex & a bit fragile & I just ran out of time to fix it with work.

In a nutshell, here's the problem:
1. The Hermitcraft-Horns clip generation server is hosted on an EC2 instance, which when I first made the app was all well & good.
2. A few months into it, the YouTube API seems to have banned the EC2 instance from making API calls (tagging it as a bot) - this means that during the "generating clip" phase, the microservice would hit YouTube's API saying "hey let me download this section of the video", but instead of YouTube sending back the clip audio it would just block the request with a "first, prove you're a human" response.
3. I started a few new EC2 instances. Every single one of them were flagged as bots from the start.
4. I started new EC2 instances in different regions. Every single one was also flagged instantly.
Ergo, it seems EC2 instances are just blocked from this API call.

I had the idea to use a VPN to wrap this, & in a sense it worked, but in another sense the network configs got so complex that in a deployed environment I had to do a bunch of manual configuration each time the server started up, & I also never got the full server to work - i.e., I was able to get YouTube's API to work, but then the actual Hermitcraft-Horns.com site lost the ability to hit the clip generation microservice.

This is really a tragedy of open-source software, I reckon. I had time when I first coded this, but not so much time anymore. Once the EC2 instance ban started hitting, I was already pretty busy with other things & none of my solutions I had time to try out were working.

I'll write into my schedule to look back into this over the next few weekends. I just can't make any promises, though. I'm really proud of the site & the community that popped up around it, so it would be a huge shame if an API block really shut the whole thing down forever.

**Comment by jrieko** [comment]
Perhaps in the meantime, an option to upload a clip file directly? For those of us able to cut the clip manually (ytdl, audacity for me). Then one could do the source video and timestamp linking as presently, but just for the metadata the site links to.

**Comment by dreadhalor** [comment]
The site is finally back up - passing the download thru a VPN seems to get around YouTube blocking the request.
